File: solutions/vbnet/sieve/1/Sieve.vb
Imports System

Public Module Sieve
  
    Public Function Primes(ByVal limit As Integer) As Integer()

        If limit <= 0 Then Throw New ArgumentOutOfRangeException

        If limit = 1 Then Return {}

        Dim primeNumbers As New List(Of Integer)

        For i = 2 To limit
            primeNumbers.Add(i)
        Next

        For i = 2 To limit
            If primeNumbers.Contains(i) Then
                For j = i * i To limit Step i
                    primeNumbers.Remove(j)
                Next
            End If
        Next

        Return primeNumbers.ToArray

    End Function

End Module

File: solutions/euphoria/sieve/1/sieve.ex
include std/sequence.e

public function primes(integer limit)
  sequence res = {}
  for i = 2 to limit do
    res &= {i}
  end for
  integer i = 0
  while i < length(res) do
    i += 1
    for j = res[i] * 2 to limit by res[i] do
      res = remove_all(j, res)
    end for
  end while
  return res
end function